src/datetime.js

```diff
@@ -199,16 +206,52 @@ function toTechTimeFormat(
     includeOffset,
     includeZone = false,
     spaceZone = false,
-    format = "extended"
+    format = "extended",
+    precision = "milliseconds"
   }
 ) {
-  let fmt = format === "basic" ? "HHmm" : "HH:mm";
-
-  if (!suppressSeconds || dt.second !== 0 || dt.millisecond !== 0) {
-    fmt += format === "basic" ? "ss" : ":ss";
-    if (!suppressMilliseconds || dt.millisecond !== 0) {
-      fmt += ".SSS";
-    }
+  // validate the precision and throw if invalid
+  let desiredPrecision = Precision[Duration.normalizeUnit(precision)];
+  if (desiredPrecision === undefined) throw new InvalidUnitError(precision);
+
+  // ---------------------------------------------------------------------------
+  // Here we adjust the precision if needed, based on whether or not
+  // 'suppressSeconds' or 'suppressMilliseconds' come into play.
+  //
+  // Note that suppressSeconds'/'suppressMilliseconds' take precedence over
+  // 'precision'. If you ask for millisecond precison, but have
+  // 'suppressSeconds' set to true, You'll get back a string with only
+  // hours and minutes when seconds and milliseconds are both zero.
```

Comment:
I think `precision` should take precedence instead. It feels more constraining, and `suppress*` is more cosmetic.

Yes, it would mean that in `{ precision : 'minutes', suppressSeconds: true }` the `suppress` part is useless, but that feels more intuitive than to display seconds when explicitly asking for a `minutes`:

`12:34:56.789  { precision : 'minutes', suppressSeconds: true } -> 12:34`
feels more natural than
`12:34:56.789  { precision : 'minutes', suppressSeconds: true } -> 12:34:56`

However,
`12:34:00  { precision : 'seconds' } -> 12:34:00`
`12:34:00  { precision : 'seconds', suppressSeconds: true } -> 12:34`

Reply:
I'll try to get to that this weekend, along with adding days, months and years.

Reply:
I agree with the above, but it also reminds me that `supressSeconds` was never a great name. `suppressSecondsIfTheyAreZero` is too long. Any ideas?

Reply:
`hideZeroSeconds` / `suppressZeroSeconds` ?

Reply:
`hideZeroSeconds` sounds good to me